enhancements/operator-hub-check.md

```diff
@@ -0,0 +1,107 @@
+
+### Goals
+
+- Perform static checks on the operator bundle manifests to avoid the scenarios where the operator bundle:
+    - is published but will not be installable
+    - is not prepared to work well on all Kubernetes cluster versions and lacks any configuration to address its limitation
+    - is using deprecated or no longer supported K8S APIs
+    - is not following good practices and conventions
+- [operator-framework/api][oper-api] with new checks in `OperatorHubValidator`
+- SDK tool using [operator-framework/api][oper-api] and return errors or warnings regarding the new checks detailed in this proposal via the command `operator-sdk bundle validate ./bundle --select-optional name=operatorhub`
+- Pipeline performing the new checks since it uses `operator-sdk bundle validate ./bundle --select-optional name=operatorhub` 
+
+### Non-Goals
+
+- Add checks which are specific for some kind of vendor
+
+## Proposal
+
+### User Stories 
+
+- As an operator developer, I'd like to use the SDK tool to validate my operator bundle before publishing on OperatorHub. Validating will allow me to quickly get its errors and warnings locally so that, I can provide the required fixes and improvements before trying to publish it.
+- As an operator framework maintainer, I'd like to see the `OperatorHubValidator` in [operator-framework/api][oper-api] performing the proposed new checks so that, I can easily keep it maintained and use it in any project and tool which pleases me.
+- As an operator framework maintainer, I'd like to see CI checking the operator bundle criteria when a user pushes a Pull Request to add the operator bundle on the OperatorHub so that, we can easily ensure a better quality of the projects before they are published.  
+
+### Implementation Details/Notes/Constraints
+
+**K8S cluster supportability**
+
+| Check | Type | Description | 
+| ------ | ----- | ----- |
+| To check if the `minKubeVersion` is informed | warning | Check if `spec.minKubeVersion` in CSV has a value if not inform the user that it would be recommended 
+| To check if the value defined in `minKubeVersion` is valid| error | Check if `spec.minKubeVersion` has an valid syntax and if it is a valid option. Otherwise, return an error with the information accordingly
+
+**CRD and Webhook APIS deprecated and unsupported versions**
+
+| Check | Type | Description | 
+| ------ | ----- | ----- |
+| To check if the operator bundle is using the CRD  API version which is deprecated | warning | Check if the CRD manifests in the bundle are using `apiVersion: apiextensions.k8s.io/v1beta1`. (See [here](https://github.com/operator-framework/operator-sdk/blob/v1.4.2/testdata/go/v2/memcached-operator/bundle/manifests/cache.example.com_memcacheds.yaml#L1) this manifest in the SDK testdata go sample using `go/v2`). Note that in these cases, users should be informed that the `apiextensions/v1beta1` was deprecated in Kubernetes `1.16` and will be removed in `1.22`.
```

Comment:
shouldn't this also be an outright error if the operator specifies a minversion of 1.22+?

Reply:
I split here. However, I think that I could found a better way to describe it.